### PR TITLE
[SPARK-30480][PYTHON][TESTS] Increases the memory limit being tested in 'WorkerMemoryTest.test_memory_limit'

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to increase the memory in `WorkerMemoryTest.test_memory_limit` in order to make the test pass with PyPy.

The test is currently failed only in PyPy as below in some PRs unexpectedly:

```
Current mem limits: 18446744073709551615 of max 18446744073709551615

Setting mem limits to 1048576 of max 1048576

RPython traceback:
  File "pypy_module_pypyjit_interp_jit.c", line 289, in portal_5
  File "pypy_interpreter_pyopcode.c", line 3468, in handle_bytecode__AccessDirect_None
  File "pypy_interpreter_pyopcode.c", line 5558, in dispatch_bytecode__AccessDirect_None
out of memory: couldn't allocate the next arena
ERROR
```

It seems related to how PyPy allocates the memory and GC works PyPy-specifically. There seems nothing wrong in this configuration implementation itself in PySpark side.

I roughly tested in higher PyPy versions on Ubuntu (PyPy v7.3.0) and this test seems passing fine so I suspect this might be an issue in old PyPy behaviours.

The change only increases the limit so it would not affect actual memory allocations. It just needs to test if the limit is properly set in worker sides. For clarification, the memory is unlimited in the machine if not set. 

### Why are the changes needed?

To make the tests pass and unblock other PRs.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Manually and Jenkins should test it out.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @HeartSaVioR, @dongjoon-hyun, @xuanyuanking

**Review by HeartSaVioR — APPROVED** [review]
LGTM. No need to be such small value actually. Thanks for dealing with this!

**Review by MaxGekk — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
retest this, please

**Review by xuanyuanking — APPROVED** [review]
LGTM, great thanks for fixing this!!
Could you also write down the current pypy version and the newer version you tested in PR description?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.
(The last commit is only `debug` to `info` change in `run-tests.py`.)

File: python/run-tests.py [hunk 1]
```diff
@@ -267,8 +267,8 @@ def main():
         python_implementation = subprocess_check_output(
             [python_exec, "-c", "import platform; print(platform.python_implementation())"],
             universal_newlines=True).strip()
-        LOGGER.debug("%s python_implementation is %s", python_exec, python_implementation)
-        LOGGER.debug("%s version is: %s", python_exec, subprocess_check_output(
+        LOGGER.info("%s python_implementation is %s", python_exec, python_implementation)
+        LOGGER.info("%s version is: %s", python_exec, subprocess_check_output(
```
Comment (HyukjinKwon) [inline python/run-tests.py]: I'm gonna check the output and piggyback this as well if it doesn't look so noisy.

**Comment by HyukjinKwon** [comment]
@xuanyuanking, IIRC, PyPy version in Jenkins is 2.5.1 which is Python 2 compatible. I fixed the log level to print out.

File: python/run-tests.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/run-tests.py]: Looks not so noisy. 6 extra lines.

```
========================================================================
Running PySpark tests
========================================================================
Running PySpark tests. Output is in /home/jenkins/workspace/SparkPullRequestBuilder/python/unit-tests.log
Will test against the following Python executables: ['python3.6', 'python2.7', 'pypy']
Will test the following Python modules: ['pyspark-core', 'pyspark-sql', 'pyspark-streaming', 'pyspark-mllib', 'pyspark-ml']
python3.6 python_implementation is CPython
python3.6 version is: Python 3.6.8 :: Anaconda, Inc.
python2.7 python_implementation is CPython
python2.7 version is: Python 2.7.13 :: Anaconda 2.1.0 (64-bit)
pypy python_implementation is PyPy
pypy version is: Python 2.7.9 (9c4588d731b7fe0b08669bd732c2b676cb0a8233, Apr 09 2015, 02:17:39)
[PyPy 2.5.1 with GCC 4.4.7 20120313 (Red Hat 4.4.7-11)]
Starting test(pypy): pyspark.sql.tests.test_arrow
...
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by xuanyuanking** [comment]
Great thanks for the guidance by @HyukjinKwon 
I locally reproduced this problem by using https://bitbucket.org/pypy/pypy/downloads/pypy-stm-2.5.1-linux64.tar.bz2. 
Here I leave more comments might helpful for future work:

- The current pypy version being used 
```
pypy version is: Python 2.7.9 (9c4588d731b7fe0b08669bd732c2b676cb0a8233, Apr 09 2015, 02:17:39)
[PyPy 2.5.1 with GCC 4.4.7 20120313 (Red Hat 4.4.7-11)]
```
seems built by ourselves? Cause  I didn't find a RedHat build on [the website](https://bitbucket.org/pypy/pypy/downloads/)
- The version 2.5.1 is super old. If more issues are found in the future, upgrade to the latest might be necessary.

**Pull request closed without merging**